### ISSUE TITLE
enhancement: Implement proper Feature Scaling pipelines for distance-based ML models

### ISSUE THREAD

**Issue by prakshithamalla-art** [Niketkumardheeryan/ML-CaPsule#1834]
### Description
**Current State:**
Several machine learning sub-modules in the repository (such as `Diabetes Prediction`, `Loan Repayment Prediction`, and standard `Classification Algorithms`) train distance-based or gradient-sensitive algorithms (like Logistic Regression, SVM, or K-Nearest Neighbors) directly on raw numerical datasets. When columns have wildly different ranges (e.g., Age 0–100 vs. Salary 0–100,000), distance-based models give unfair weight to larger numbers, leading to skewed predictions.

**Proposed Solution:**
Introduce proper data preprocessing using Scikit-Learn's `StandardScaler` or `MinMaxScaler` to normalize features before they are passed into the classification models.

**Implementation Steps:**
1. Target 1-2 folders dealing with numerical datasets (e.g., Diabetes or Loan Prediction).
2. Integrate a feature scaling step directly after splitting the data into training and testing sets:
   ```python
   from sklearn.preprocessing import StandardScaler

   scaler = StandardScaler()
   X_train_scaled = scaler.fit_transform(X_train)
   X_test_scaled = scaler.transform(X_test)

**Comment by suvancodes** [comment]
Hi @prakshithamalla-art ,

I am a GSSoC 2026 participant and would like to contribute to this enhancement. I have experience with Python, Scikit-Learn, and Machine Learning workflows, and I understand how feature scaling impacts distance-based models.

I would be happy to implement the proposed solution and submit a well-tested PR. Kindly assign this issue to me.

Thanks! 🚀
